tests/integration/test_save_load_error_handling.py

```
@@ -0,0 +1,216 @@
+        save_path = os.path.join(self.temp_dir, "test_bot.bot")
+        bot.save(save_path)
+
+        # Modify the saved file to trigger code hash mismatch
+        import json
+
+        with open(save_path, "r") as f:
+            data = json.load(f)
+
+        # Change the code hash to force mismatch
+        if "tool_handler" in data and "modules" in data["tool_handler"]:
+            for module_path, module_data in data["tool_handler"]["modules"].items():
```

Ruff (0.14.1)

Loop control variable `module_path` not used within loop body

Rename unused `module_path` to `_module_path`

(B007)